buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/IntellijPlugin.kt

```diff
@@ -85,8 +88,71 @@ fun Project.configureIntellijPlugin() {
     dependsOn(verifyVersionTask, verifyContentsTask)
   }
 
+  tasks.register("printPluginZipPath") {
+    doLast {
+      val buildPlugin = tasks.findByPath(":buildPlugin")!!
+      println(buildPlugin.outputs.files.first().path)
+    }
+  }
+
+  tasks.register("printSignedPluginZipPath") {
+    // Required to prevent https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1358
+    dependsOn(":buildPlugin")
+
+    doLast {
+      val signPlugin = tasks.findByPath(":signPlugin")!!
+      println(signPlugin.outputs.files.first().path)
+    }
+  }
+
+  val verifyPluginZipTask = tasks.register("verifyPluginZip") {
+    val buildPlugin = tasks.findByPath(":buildPlugin")!!
+    dependsOn(buildPlugin)
+
+    doLast {
+      val pluginZipPath = buildPlugin.outputs.files.first().path
+      val jarsInPluginZip = ZipFile(pluginZipPath).use { zf ->
+        zf.stream()
+          .map(ZipEntry::getName)
+          .map { it.removePrefix("git-machete-intellij-plugin/").removePrefix("lib/").removeSuffix(".jar") }
+          .filter { it.isNotEmpty() }
+          .toList()
+      }
+
+      for (proj in subprojects) {
+        val projJar = proj.path.replaceFirst(":", "").replace(":", "-")
+        val javaExtension = proj.extensions.getByType<JavaPluginExtension>()
+        if (javaExtension.sourceSets["main"].allSource.srcDirs.any { it?.exists() == true }) {
+          check(projJar in jarsInPluginZip) {
+            "$projJar.jar was expected in plugin zip ($pluginZipPath) but was NOT found"
+          }
+        } else {
+          check(projJar !in jarsInPluginZip) {
+            "$projJar.jar was NOT expected in plugin zip ($pluginZipPath) but was found"
+          }
+        }
+      }
+
+      val expectedLibs = listOf("org.eclipse.jgit", "slf4j-lambda-core", "vavr", "vavr-match")
+      for (expectedLib in expectedLibs) {
+        val libRegex = "^" + expectedLib.replace(".", "\\.") + "-[0-9.]+.*$"
```

Comment:
iiuc technically at this point the value is more like `libRegexStr` and it becomes a regex after `toRegex()` call 🤔